src/variant.rs

```diff
@@ -378,10 +475,164 @@ impl<T: ToVariant> From<T> for Variant {
     }
 }
 
+impl<T: StaticVariantType> StaticVariantType for [T] {
+    fn static_variant_type() -> Cow<'static, VariantTy> {
+        let child_type = T::static_variant_type();
+        let signature = format!("a{}", child_type.to_str());
+
+        VariantType::new(&signature).expect("incorrect signature").into()
+    }
+}
+
+impl<T: FromVariant> FromVariant for Vec<T> {
+    fn from_variant(variant: &Variant) -> Option<Self> {
+        let mut vec = vec![];
+
+        let n_children = variant.n_children();
```

Comment:
If there's no `append_to_array()` API possible, you could do `Vec::with_capacity(variant.n_children())` at least to prevent reallocations